.github/workflows/main.yml

```diff
@@ -7,30 +7,56 @@ on:
   push
 
 jobs:
-  build:
-
+  test:
     runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]  # 3.9 not supported on numba as of 2020-02-11
     steps:
-
-      - uses: actions/checkout@v2
-      - name: Set up Python 3.x
-        uses: actions/setup-python@v2
-        with:
-          # Semantic version range syntax or exact version of a Python version
-          python-version: '3.7'
-
-      - name: Install dependencies
-        shell: bash -l {0}
-        run: |
-          echo Branch: ${GITHUB_REF}
-          pip install pytest
-          python -m pip install --upgrade pip
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-      - name: Test with pytest
-        shell: bash -l {0}
-        run: |
-          ls
-          pwd
-          export PYTHONPATH="$PYTHONPATH:$(pwd)"
-          pytest
+    - name: Checkout PyAutoFit
+      uses: actions/checkout@v2
+      with:
+        repository: rhayes777/PyAutoFit
+        path: PyAutoFit
+    - name: Checkout PyAutoArray
+      uses: actions/checkout@v2
+      with:
+        repository: Jammy2211/PyAutoArray
+        path: PyAutoArray
+    - name: Checkout PyAutoGalaxy
+      uses: actions/checkout@v2
+      with:
+        repository: Jammy2211/PyAutoGalaxy
+        path: PyAutoGalaxy
+    - name: Checkout PyAutoLens
+      uses: actions/checkout@v2
+      with:
+        repository: Jammy2211/PyAutoLens
+        path: PyAutoLens
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - uses: actions/cache@v2
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+    - name: Install dependencies
+      if: steps.cache-pip.outputs.cache-hit != 'true'
+      run: |
+        pip install -r PyAutoFit/requirements.txt
+        pip install -r PyAutoArray/requirements.txt
+        pip install -r PyAutoGalaxy/requirements.txt
```

Comment:
AutoArray doesn't depend on Galaxy and Lens, should they be in this workflow script?